cpp/src/barretenberg/proof_system/composer/permutation_helper.hpp

```diff
@@ -456,82 +495,11 @@ template <size_t program_width, typename CircuitConstructor>
-    for (size_t i = 0; i < program_width; ++i) {
-        sigma_mappings[i].reserve(key->circuit_size);
-        id_mappings[i].reserve(key->circuit_size);
-    }
-    for (size_t i = 0; i < program_width; ++i) {
-        for (size_t j = 0; j < key->circuit_size; ++j) {
-            sigma_mappings[i].emplace_back(permutation_subgroup_element{
-                .row_index = (uint32_t)j, .column_index = (uint8_t)i, .is_public_input = false, .is_tag = false });
-
-            id_mappings[i].emplace_back(permutation_subgroup_element{
-                .row_index = (uint32_t)j, .column_index = (uint8_t)i, .is_public_input = false, .is_tag = false });
-        }
-    }
-
-    // // Represents the index of a variable in circuit_constructor.variables
-    std::span<const uint32_t> real_variable_tags = circuit_constructor.real_variable_tags;
-    // const std::map<uint32_t, uint32_t>& tau = circuit_constructor.tau;
-
-    // Go through all wire cycles and update sigma and id mappings to point to the next element
-    // within each cycle as well as set the appropriate tags
-    for (size_t i = 0; i < wire_copy_cycles.size(); ++i) {
-        for (size_t j = 0; j < wire_copy_cycles[i].size(); ++j) {
-            cycle_node current_cycle_node = wire_copy_cycles[i][j];
-            size_t next_cycle_node_index = j == wire_copy_cycles[i].size() - 1 ? 0 : j + 1;
-            cycle_node next_cycle_node = wire_copy_cycles[i][next_cycle_node_index];
-            const auto current_row = current_cycle_node.gate_index;
-            const auto next_row = next_cycle_node.gate_index;
-
-            const uint32_t current_column = current_cycle_node.wire_index;
-            const uint32_t next_column = next_cycle_node.wire_index;
-
-            sigma_mappings[current_column][current_row] = {
-                .row_index = next_row, .column_index = (uint8_t)next_column, .is_public_input = false, .is_tag = false
-            };
-
-            bool first_node, last_node;
-
-            first_node = j == 0;
-            last_node = next_cycle_node_index == 0;
-            if (first_node) {
-                id_mappings[current_column][current_row].is_tag = true;
-                id_mappings[current_column][current_row].row_index = (real_variable_tags[i]);
-            }
-            if (last_node) {
-                sigma_mappings[current_column][current_row].is_tag = true;
-
-                // TODO: yikes, std::maps are expensive. Can we find a way to get rid of this?
-                sigma_mappings[current_column][current_row].row_index =
-                    circuit_constructor.tau.at(real_variable_tags[i]);
-            }
-        }
-    }
-
-    const uint32_t num_public_inputs = static_cast<uint32_t>(circuit_constructor.public_inputs.size());
-
-    // This corresponds in the paper to modifying sigma to sigma' with the zeta_i values; this enforces public input
-    // consistency
-    for (size_t i = 0; i < num_public_inputs; ++i) {
```

Comment:
Where did this logic go?

Reply:
This takes place in the final lines of `compute_permutation_mapping`